client/webpack/generate-chunks-map-plugin.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies.
+ */
+const fs = require( 'fs' ); //eslint-disable-line import/no-nodejs-modules
+const path = require( 'path' );
+
+/**
+ * Plugin name.
+ *
+ * @type {string}
+ */
+const PLUGIN_NAME = 'GenerateChunksMap';
+
+class GenerateChunksMapPlugin {
+	constructor( { output = path.resolve( '.', 'map.json' ) } = {} ) {
+		this.output = output;
+	}
+
+	apply( compiler ) {
+		compiler.hooks.done.tap( PLUGIN_NAME, ( { compilation } ) => {
+			const { chunks } = compilation;
+
+			const chunksMap = chunks.reduce( ( map, chunk ) => {
+				const files = chunk.files || [];
+				const name = files.find( file => /\.js$/.test( file ) ) || files[ 0 ];
```

Comment:
I don't think this is going to generate a unique name. Any given file could appear in more than one chunk. It's unlikely, but very possible.

Reply:
Also, given you're defaulting to `[]`, `name` could be undefined, which isn't a valid map key for the next line.

Reply:
Maybe instead of building the name from the filename, we could use the chunkhash?

Reply:
`chunk.files` contains the .js file name and in some cases its .css and .rtl.css that each chunk will output, not the individual modules it includes. So, it should be safe to use.

As for the `[]` fallback value - seems like I forgot to remove it after debugging something else. It's not really needed, since each chunk has atleast its .js files.